Client#submit_tx.: Fix Base64 encoding

joallard opened the pull request:
Without strict, produces newline characters which Horizon will reject

---

(Split from #95:)

And so I was getting some TransactionMalformed errors. Looks like this is because Base64.encode64 by default inserts some newlines, which Horizon doesn't like.

The tests probably didn't catch this because of some combination of using VCR and upgrades on our side.

nebolsin commented on sdk/lib/stellar/client.rb:
```diff
@@ -199,7 +199,7 @@ def submit_transaction(tx_envelope:, options: {skip_memo_required_check: false})
         check_memo_required(tx_envelope)
       end
-      @horizon.transactions._post(tx: Base64.encode64(tx_envelope.to_xdr))
+      @horizon.transactions._post(tx: Base64.strict_encode64(tx_envelope.to_xdr))
```
```suggestion
      @horizon.transactions._post(tx: tx_envelope.to_xdr(:base64))
```

I think we could simplify this, since `ruby-xdr` has an option to serialize into base64, and it correctly uses `strict_encode64`: https://github.com/astroband/ruby-xdr/blob/master/lib/xdr/concerns/converts_to_xdr.rb#L50.

joallard replied:
Makes a lot of sense to me!

joallard replied:
I never expected to get a good review on such a small diff :joy: Well done!

joallard committed:
Client#submit_tx: Fix Base64 encoding by using envelope.to_xdr(:base64)

Ruby's standard Base64.encode64 inserts newline separators by default, 
which Horizon doesn't accept.

nebolsin reviewed (approved):
:lgtm:
